src/main/java/com/mparticle/kits/SwrveKit.java

```diff
@@ -0,0 +1,371 @@
+ *
+ *
+ * Follow the steps below to implement your kit:
+ *
+ *  - Edit ./build.gradle to add any necessary dependencies, such as your company's SDK
+ *  - Rename this file/class, using your company name as the prefix, ie "AcmeKit"
+ *  - View the javadocs to learn more about the KitIntegration class as well as the interfaces it defines.
+ *  - Choose the additional interfaces that you need and have this class implement them,
+ *    ie 'AcmeKit extends KitIntegration implements KitIntegration.PushListener'
+ *
+ *  In addition to this file, you also will need to edit:
+ *  - ./build.gradle (as explained above)
+ *  - ./README.md
+ *  - ./src/main/AndroidManifest.xml
+ *  - ./consumer-proguard.pro
+ */
+public class SwrveKit extends KitIntegration implements KitIntegration.UserAttributeListener, KitIntegration.CommerceListener, KitIntegration.EventListener, KitIntegration.PushListener, KitIntegration.IdentityListener {
+    private static final String SWRVE_MPARTICLE_VERSION_NUMBER = "1.0.0";
+
+    private void startSwrveSDK(Activity activity, long mpid) {
+        SwrveSDK.start(activity, Long.toString(mpid));
+        Map<String,String> version = new HashMap<String,String>();
+        version.put("swrve.mparticle_android_integration_version", SWRVE_MPARTICLE_VERSION_NUMBER);
+        SwrveSDK.userUpdate(version);
+        SwrveSDK.sendQueuedEvents();
+    }
+
+    @Override
+    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
+        /** TODO: Initialize your SDK here
+         * This method is analogous to Application#onCreate, and will be called once per app execution.
+         *
+         * If for some reason you can't start your SDK (such as settings are not present), you *must* throw an Exception
+         *
+         * If you forward any events on startup that are analagous to any mParticle messages types, return them here
+         * as ReportingMessage objects. Otherwise, return null.
+         */
+        List<ReportingMessage> messageList = new LinkedList<ReportingMessage>();
+        SwrveConfig config = new SwrveConfig();
+        SwrveNotificationConfig notificationConfig = getNotificationConfig(settings, context);
+        config.setNotificationConfig(notificationConfig);
+
+        config.setInitMode(SwrveInitMode.MANAGED);
+        // TODO when MParticleUser and userId is known, call SwrveSDK.start(activity, userId). Do not call start from application layer.
+
+        int app_id = Integer.parseInt(settings.get("app_id"));
+        String api_key = settings.get("api_key");
+        // To use the EU stack, include this in your config.
+        // config.setSelectedStack(SwrveStack.EU);
+        SwrveSDK.createInstance( ( (Application) context.getApplicationContext() ), app_id, api_key, config);
+        messageList.add(new ReportingMessage(this, ReportingMessage.MessageType.SESSION_START, System.currentTimeMillis(), null));
```

Comment:
you can remove this, afaik Swrve isn't actually consuming the Session Start, so we don't need to return a ReportingMessage. You can either return `null` or an empty list